### ISSUE TITLE
MacOS environment now provides a /usr/local/bin/2to3, conflicting with homebrew/homebrew dependencies

### ISSUE THREAD

**Issue by fredemmott** [actions/virtual-environments#4033]
### Description

For example, https://github.com/hhvm/hh-clilib/runs/3526321802?check_suite_focus=true : the hhvm package has a transitive dependency on the python@3.9 package:

```
 ==> Installing hhvm/hhvm/hhvm-4.124 dependency: python@3.9
  ==> Pouring python@3.9--3.9.7.catalina.bottle.tar.gz
  Error: The `brew link` step did not complete successfully
  The formula built, but is not symlinked into /usr/local
  Could not symlink bin/2to3
  Target /usr/local/bin/2to3
  already exists. You may want to remove it:
    rm '/usr/local/bin/2to3'
```

Notably:
- this appears to be a regression of #2322
- there is explicit handling of this in https://github.com/actions/virtual-environments/blob/b7f276c003aea42575b52247bdb2183e355fca2f/images/macos/provision/core/python.sh#L15 which appears to not be working
- this worked fine last week, and doesn't work today
- the image version has not changed since last successful build
- I'm also seeing the same change over the long weekend on raw azure pipelines
- `brew update; brew upgrade` on my local mac (intel, macos 11.5.2) has no problems

This could be a change to homebrew triggering this.

### Virtual environments affected

- [ ] Ubuntu 16.04
- [ ] Ubuntu 18.04
- [ ] Ubuntu 20.04
- [X] macOS 10.15
- [ ] macOS 11
- [ ] Windows Server 2016
- [ ] Windows Server 2019
- [ ] Windows Server 2022

### Image version and build link

  Environment: macos-10.15
  Version: 20210831.3
  
  https://github.com/hhvm/hh-clilib/runs/3526321802

### Is it regression?

Yes - same image version though: https://github.com/hhvm/hh-clilib/runs/3513952202?check_suite_focus=true

### Expected behavior

python 3.9 installed without errors

### Actual behavior

```
  ==> Pouring python@3.9--3.9.7.catalina.bottle.tar.gz
  Error: The `brew link` step did not complete successfully
  The formula built, but is not symlinked into /usr/local
  Could not symlink bin/2to3
  Target /usr/local/bin/2to3
  already exists. You may want to remove it:
    rm '/usr/local/bin/2to3'
```

### Repro steps

`brew tap hhvm/hhvm; brew install hhvm-4.124`

**Comment by catthehacker** [comment]
Duplicate of https://github.com/actions/virtual-environments/issues/4020

**Comment by fredemmott** [comment]
Also appears fixed by 20210905.2 image ( https://github.com/hhvm/hh-clilib/runs/3535625772?check_suite_focus=true )
